source/ota.c

```diff
@@ -1941,65 +1930,59 @@ static OtaJobParseErr_t handleCustomJob( const char * pJson,
     jobDoc.parseErr = OtaJobParseErrUnknown;
 
     /* If this is a valid custom job, extract job document and job ID data from the JSON payload.*/
-    if( JSONSuccess != JSON_SearchConst( pJson,
-                                         messageLength,
-                                         OTA_JSON_JOB_DOC_KEY,
-                                         strlen( OTA_JSON_JOB_DOC_KEY ),
-                                         ( const char ** ) &jobDoc.pJobDocJson,
-                                         &jobDoc.jobDocLength,
-                                         NULL ) )
-    {
-        if( JSONSuccess == JSON_SearchConst( pJson,
-                                             messageLength,
-                                             OTA_JSON_JOB_ID_KEY,
-                                             strlen( OTA_JSON_JOB_ID_KEY ),
-                                             ( const char ** ) &jobDoc.pJobId,
-                                             &jobDoc.jobIdLength,
-                                             NULL ) )
-        {
-            /* We have an unknown job parser error. Check to see if we can pass control
-             * to a callback for parsing */
-            otaAgent.OtaAppCallback( OtaJobEventParseCustomJob, &jobDoc );
+    if( ( JSONSuccess == JSON_SearchConst( pJson,
+                                           messageLength,
+                                           OTA_JSON_JOB_DOC_KEY,
+                                           strlen( OTA_JSON_JOB_DOC_KEY ),
+                                           ( const char ** ) &jobDoc.pJobDocJson,
+                                           &jobDoc.jobDocLength,
+                                           NULL ) ) &&
+
+        ( JSONSuccess == JSON_SearchConst( pJson,
+                                           messageLength,
+                                           OTA_JSON_JOB_ID_KEY,
+                                           strlen( OTA_JSON_JOB_ID_KEY ),
+                                           ( const char ** ) &jobDoc.pJobId,
+                                           &jobDoc.jobIdLength,
+                                           NULL ) ) )
+    {
+        /* We have an unknown job parser error. Check to see if we can pass control
+         * to a callback for parsing */
+        otaAgent.OtaAppCallback( OtaJobEventParseCustomJob, &jobDoc );
+
+        if( jobDoc.parseErr == OtaJobParseErrNone )
```

Comment:
It looks like we're introducing 2 new tests (lines 1953 & 1962), but I don't see unit test cases added for these in this PR.  Can we add them?